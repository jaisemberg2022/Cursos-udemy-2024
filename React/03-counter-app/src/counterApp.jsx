import {useState} from 'react'
import { Button, Stack, Typography,ButtonGroup } from "@mui/material";
import React from "react";

export const CounterApp = ({ value }) => {
    
    const [counter,setCounter] = useState(value)

    const handleAdd = () => setCounter(counter +1);
    const hanndleSubstract =()=> setCounter(counter -1);
    const hanndleReset =()=> setCounter(value);
    

  return (
    <>
      <Stack p={2} display={"inline-flex"} gap={2} width={'auto'} boxShadow={2} alignItems={"center"} flexDirection={'column'} m={2}>
        <Typography variant="h4">CounterApp</Typography>
        <h6>{counter}</h6>
        <Stack width={'100%'} display={'inline-flex'} gap={1} flexDirection={'row'}>
             <ButtonGroup variant="outlined">
                <Button size='small' onClick={ handleAdd }>+1</Button>
                <Button onClick={hanndleReset}>restore</Button>
                <Button onClick={hanndleSubstract}>-1</Button>
            </ButtonGroup>
        </Stack>
      </Stack>
    </>
  );
};

import PropTypes from "prop-types";

CounterApp.propTypes = {
  value: PropTypes.number,
};
