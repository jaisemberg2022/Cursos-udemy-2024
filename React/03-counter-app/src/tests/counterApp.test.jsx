import {render,screen} from '@testing-library/react';
import { CounterApp } from '../counterApp';

describe('pruebas en <counterapp/>', () => { 
    const value = 10;
    test('Debe hacer match con snapshot',()=>{
        const {container} = render(
            <CounterApp value={value}/>
        );
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el valor en 100',()=>{
        render(
            <CounterApp value={100}/>
        );
        expect(screen.getByText(100)).toBeTruthy();
    })
 })