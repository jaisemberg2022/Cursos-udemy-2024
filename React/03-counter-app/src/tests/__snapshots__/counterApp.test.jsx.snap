// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <counterapp/> Debe hacer match con snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1dyvdz9-MuiStack-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      CounterApp
    </h4>
    <h6>
      10
    </h6>
    <div
      class="MuiStack-root css-prnffn-MuiStack-root"
    >
      <div
        class="MuiButtonGroup-root MuiButtonGroup-outlined css-edsu1d-MuiButtonGroup-root"
        role="group"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-firstButton css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          +1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-middleButton css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          restore
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-lastButton css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          -1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
